src/components/NavbarForm.jsx

```diff
@@ -0,0 +1,27 @@
+import React from 'react';
+import PropTypes from 'prop-types';
+import {Navbar} from 'react-bootstrap';
+import {connect} from 'react-redux';
+import LoginForm from './LoginForm'
+
+export class NavbarForm extends React.Component {
+  render() {
+    return (
+        <Navbar.Form pullRight >
+          <LoginForm toggleCollapse={this.context.$bs_navbar.onToggle}/>
+        </Navbar.Form >
+    )
+  }
+}
+
+NavbarForm.contextTypes = {
+  $bs_navbar: PropTypes.shape({
+    onToggle: PropTypes.func.isRequired
```

Comment:
Despite having to declare the contextTypes shape, you can limit the declaration to what you actually need.